{{cookiecutter.project_slug}}/.pre-commit-config.yaml

```diff
@@ -0,0 +1,32 @@
+exclude: .+/migrations/.+\.py
+default_stages: [commit]
+fail_fast: true
+
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: master
+    hooks:
+      - id: trailing-whitespace
+        files: (^|/){{cookiecutter.project_slug}}/.+\.(py|html|sh|css|js)$
+
+-   repo: local
+    hooks:
+      - id: black
+        name: black
+        entry: black
+        args: [--line-length=80]
+        language: python
+        types: [python]
+        files: (^|/){{cookiecutter.project_slug}}/.+\.py$
+        verbose: true
+
+-   repo: local
+    hooks:
+      - id: flake8
+        name: flake8
+        entry: flake8
+        args: [--max-line-length=120]
```

Comment:
If we don't specify this, would the hook use the project config from the `setup.cfg`? That would be nice to avoid duplication...

https://github.com/pydanny/cookiecutter-django/blob/19eb42a7cf1bdf41fa3deef0af976ba2c7f2d1d3/%7B%7Bcookiecutter.project_slug%7D%7D/setup.cfg#L2

As a side note, I realised that the flake8 config should probably be updated to be [in line with Black's](https://black.readthedocs.io/en/stable/the_black_code_style.html#line-length) (out of scope of this pull request, but feel free to update it).

Reply:
I am going to remove `args`, meanwhile update max-line-length to 88 for flake8 config in setup.cfg.

Reply:
It can't be set to 88.

there is a one line whose length is greater than 88.
[config/settings/base.py](https://github.com/pydanny/cookiecutter-django/blob/19eb42a7cf1bdf41fa3deef0af976ba2c7f2d1d3/%7B%7Bcookiecutter.project_slug%7D%7D/config/settings/base.py#L122)

or, it has to break like this:
```
AUTH_PASSWORD_VALIDATORS = [
      {
          "NAME": "django.contrib.auth.password_validation.\
                  UserAttributeSimilarityValidator"
      },
      {"NAME": "django.contrib.auth.password_validation.MinimumLengthValidator"},
      {"NAME": "django.contrib.auth.password_validation.CommonPasswordValidator"},
      {"NAME": "django.contrib.auth.password_validation.NumericPasswordValidator"},
  ]
```